src/com/oltpbenchmark/api/collectors/MyRocksCollector.java

```diff
@@ -61,25 +84,60 @@ public MyRocksCollector(String oriDBUrl, String username, String password) {
             // Collect MyRocks Column Family parameters
             out = s.executeQuery(CF_OPTIONS);
             while(out.next()){
-                dbParameters.put(out.getString(1).toLowerCase() + " | " + out.getString(2).toLowerCase(), out.getString(3));
+                dbParameters.put("cf_option: " + "cf_name=" + out.getString(1).toLowerCase() + " , " + "type=" + out.getString(2).toLowerCase(), out.getString(3));
             }
 
             // Collect DBMS internal metrics
             out = s.executeQuery(METRICS_SQL);
-            while (out.next()) {
-            	dbMetrics.put(out.getString(1).toLowerCase(), out.getString(2));
-            }
+            myroMetrics.put("internal_metrics", getMetrics(out));
+                   
+            // Collect metrics from information_schema 
+	    for (String viewName : MYRO_STAT_VIEWS) {
+            	out = s.executeQuery("select * from information_schema." + viewName + ";");
+            	myroMetrics.put(viewName, getMetrics(out));
+            }          
+ 
+            // Collect db statistics
+            out = s.executeQuery(DB_STATS + "\""+dbname +"\""+ ";");
+            myroMetrics.put("db_statistics", getMetrics(out));
+           
+	    // Collect table statistics  
+            out = s.executeQuery(TABLE_STATS + "\""+dbname +"\""+ ";");
+            myroMetrics.put("table_statistics", getMetrics(out));
+
+	    // Collect index statistics  
+            out = s.executeQuery(INDEX_STATS + "\""+dbname +"\""+ ";");
+            myroMetrics.put("index_statistics", getMetrics(out));
 
-            // Collect MyRocks Column Family statistics 
-            out = s.executeQuery(CF_STATS);
-            while(out.next()){
-                dbMetrics.put(out.getString(1).toLowerCase() + " | " + out.getString(2).toLowerCase(), out.getString(3));
-            }            
+        } catch (SQLException e) {
+            LOG.error("Error while collecting DB parameters: " + e.getMessage());
+        }
+    }
 
+    @Override
+    public String collectMetrics() {
+    	return JSONUtil.format(JSONUtil.toJSONString(myroMetrics));
+    }  
 
 
-        } catch (SQLException e) {
-            LOG.error("Error while collecting DB parameters: " + e.getMessage());
+    private static List<Map<String, String>> getMetrics(ResultSet out) throws SQLException {
```

Comment:
This is the exact same method that's in the PostgresCollector, right? Can you pull this method out and put it in DBCollector? Then call it from both the PostgresCollector & MyRocksCollector.